packages/teams-js/src/private/externalAppAuthenticationForCEA.ts

```diff
@@ -0,0 +1,269 @@
+   */
+  export async function authenticateWithSSOAndResendRequest(
+    appId: AppId,
+    conversationId: string,
+    authTokenRequest: externalAppAuthentication.AuthTokenRequestParameters,
+    originalRequestInfo: externalAppAuthentication.IActionExecuteInvokeRequest,
+  ): Promise<externalAppAuthentication.IActionExecuteResponse> {
+    ensureInitialized(runtime, FrameContexts.content);
+
+    if (!isSupported()) {
+      throw errorNotSupportedOnPlatform;
+    }
+
+    validateId(conversationId, new Error('conversation id is not valid.'));
+
+    validateOriginalRequestInfo(originalRequestInfo);
+
+    const [response] = await sendMessageToParentAsync<
+      [externalAppAuthentication.IActionExecuteResponse | externalAppAuthentication.InvokeErrorWrapper]
+    >(
+      getApiVersionTag(
+        externalAppAuthenticationTelemetryVersionNumber,
+        ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSOAndResendRequest,
+      ),
+      ApiName.ExternalAppAuthenticationForCEA_AuthenticateWithSSOAndResendRequest,
+      [appId.toString(), conversationId, originalRequestInfo, authTokenRequest.claims, authTokenRequest.silent],
+    );
+    if (isActionExecuteResponse(response)) {
+      return response;
+    } else if (isInvokeErrorWrapper(response)) {
+      throw response;
+    } else {
+      throw defaultExternalAppError;
+    }
+  }
+
+  /**
+   * @beta
+   * @hidden
+   * Checks if the externalAppAuthenticationForCEA capability is supported by the host
+   * @returns boolean to represent whether externalAppAuthenticationForCEA capability is supported
+   * @throws Error if {@linkcode app.initialize} has not successfully completed
+   * @internal
+   * Limited to Microsoft-internal use
+   */
+  export function isSupported(): boolean {
+    return ensureInitialized(runtime) && runtime.supports.externalAppAuthenticationForCEA ? true : false;
+  }
+
+  /**
+   * @hidden
+   * @internal
+   * Limited to Microsoft-internal use
+   * @beta
+   */
+  function validateOriginalRequestInfo(
+    actionExecuteRequest: externalAppAuthentication.IActionExecuteInvokeRequest,
+  ): void {
+    if (actionExecuteRequest.type !== externalAppAuthentication.ActionExecuteInvokeRequestType) {
+      const error: externalAppAuthentication.InvokeError = {
+        errorCode: externalAppAuthentication.InvokeErrorCode.INTERNAL_ERROR,
+        message: `Invalid action type ${actionExecuteRequest.type}. Action type must be "${externalAppAuthentication.ActionExecuteInvokeRequestType}"`,
+      };
+      throw error;
+    }
+  }
+
+  function isInvokeErrorWrapper(err: unknown): err is externalAppAuthentication.InvokeErrorWrapper {
+    if (typeof err !== 'object' || err === null) {
+      return false;
+    }
+
+    const errorWrapper = err as externalAppAuthentication.InvokeErrorWrapper;
+
+    return (
+      errorWrapper?.errorCode === externalAppAuthentication.InvokeErrorCode.INTERNAL_ERROR &&
```

Comment:
>errorWrapper?.errorCode === externalAppAuthentication.InvokeErrorCode.INTERNAL_ERROR [](http://example.com/codeflow?start=6&length=84)

I don't think it's right to test that `errorWrapper.errorCode` is exactly this value, since it could be some other value in the future? The `errorCode` property can be any `InvokeErrorCode` value. Yes, it is true that `InvokeErrorCode` has only one value right now, but it can have more in the future. If this code only tests for `InvokeErrorCode.INTERNAL_ERROR`, then this function will no longer work when new error codes are added (is the expectation that every new `InvokeErrorCode` gets added to the `enum` and also updated here?